File: API/Entities/Expenses.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities
{
    public class Expenses:BaseEntity
    {
        [Required]
        public DateTime Date { get; set; }
        [Required]
        public Guid CategoryId { get; set; }
        public virtual Category Category { get; set;}
        public string Currency { get; set; }
        public decimal Amount { get; set; }
        public Guid? TaxId { get; set; }
        public virtual Tax Tax { get; set; }
        public string ReferenceNumber { get; set; }
        public string Notes { get; set; }
        public Guid? CustomerId { get; set; }
        public virtual Customer Customer { get; set; }
       

    }
}

File: Client/src/app/_model/expenses.ts
import { Guid } from 'guid-typescript';
import { Category } from './category';
import { Customer } from './customer';
import { Tax } from './tax';

export interface Expenses {
  id: Guid;
  date: Date;
  category: Category;
  currency: string;
  amount: Number;
  tax: Tax;
  referenceNumber: string;
  notes: string;
  customer: Customer;
}
